addons/mail/static/src/model/model_manager.js

```diff
@@ -235,31 +305,44 @@ class ModelManager {
         return this.all(Model).find(findFunc);
     }
 
+    /**
+     * Get the record of provided model that has provided identifying data,
+     * if it exists.
+     *
+     * @param {mail.model} Model class
+     * @param {Object} data
+     * @returns {mail.model|undefined} the record of model identified by data,
+     *  if it exists.
+     */
+    findFromData(Model, data) {
+        const index = this._identifyingKey(Model, data);
+        return this._models.get(Model).indexes.primary[index];
+    }
+
     /**
      * This method returns the record of provided model that matches provided
-     * record/local id. Useful to convert a local id to a record, and also to
-     * determine whether the record is still "alive" (i.e. not deleted). Note
-     * that even if there's a record in the system having provided local id, if
-     * the resulting record is not an instance of this model, this getter
+     * local id. Useful to convert a local id to a record.
+     * Note that even if there's a record in the system having provided local
+     * id, if the resulting record is not an instance of this model, this getter
      * assumes the record does not exist.
      *
      * @param {mail.model} Model class
-     * @param {string|mail.model|undefined} recordOrLocalId
+     * @param {string} localId
      * @returns {mail.model|undefined} record, if exists
      */
-    get(Model, recordOrLocalId) {
-        if (recordOrLocalId === undefined) {
-            return undefined;
-        }
-        const record = this._records[
-            recordOrLocalId instanceof this.env.models['mail.model']
-                ? recordOrLocalId.localId
-                : recordOrLocalId
-        ];
-        if (!(record instanceof Model)) {
+    get(Model, localId) {
+        if (!localId) {
             return;
         }
-        return record;
+        if (Model.modelName === 'mail.model') {
+            // special case necessary for relations targeting generic model
+            for (const model of this._models.values()) {
+                if (localId in model.records) {
+                    return model.records[localId];
+                }
+            }
+        }
+        return this._models.get(Model).records[localId];
```

Comment:
This doesn't work if we have inheritance on custom models.

Reply:
Ok, removed condition on model name, and loop everywhere only if no direct match is found.